Always enable --single-version-externally-managed

This ensures packages are never installed in zipfiles and makes sure --record
or --root is not required with --single-version-externally-managed.  It means
any conda recipes using this patched setuptools do not have to worry about using
these options.

--- setuptools/command/install.py.orig
+++ setuptools/command/install.py
@@ -32,18 +32,10 @@
     def initialize_options(self):
         orig.install.initialize_options(self)
         self.old_and_unmanageable = None
-        self.single_version_externally_managed = None
+        self.single_version_externally_managed = True
 
     def finalize_options(self):
         orig.install.finalize_options(self)
-        if self.root:
-            self.single_version_externally_managed = True
-        elif self.single_version_externally_managed:
-            if not self.root and not self.record:
-                raise DistutilsArgError(
-                    "You must specify --record or --root when building system"
-                    " packages"
-                )
 
     def handle_extra_path(self):
         if self.root or self.single_version_externally_managed:
